### PR TITLE
GitHub Action for unit tests

### PR DESCRIPTION
**Pull request by x-way**
Adds a GitHub Action to run unit tests and code linting.
Linting is only run on Linux, but unit tests are run for Linux, Mac and Windows (so we cover our target platform 🙂).

Fixed the warnings from the linters (mostly missing comments on exported functions and use of some deprecated APIs).

Comes with a simple unit test for the config parsing, the more interesting parts for testing will be the WiresX log parsing.

### CODE REVIEW

**Review by roger-dodger — COMMENTED** [review]
Looks good for me. Looking forward for the log parsing unit tests on Windows ;-)

File: config/config.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// Package config provides configuration reading functionality
```
Comment (finfinack) [inline config/config.go]: Thanks for adding comments!

Nit: Please add periods at the end of the comments/sentences.

File: config/config_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			got, err := config.Read(tc.inputfile)
+			if err != nil {
+				t.Errorf("Read('%s') returned an error: %v", tc.inputfile, err)
+			}
+			if diff := deep.Equal(tc.expected, got); diff != nil {
```
Comment (finfinack) [inline config/config_test.go:47]: Not actionable but FYI, I usually use cmp.Diff in these cases:
https://pkg.go.dev/github.com/google/go-cmp/cmp#example-Diff-Testing

(but mostly out of habit, not sure if one has a real advantage over the other to be honest)

File: config/config.go (hunk 1 above)
Comment (finfinack) [reply config/config.go]: (also in the other files with comments you added)

**Pull request merged**